pkg/condition/condition.go

```diff
@@ -0,0 +1,109 @@
+package condition
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"time"
+
+	operators "github.com/operator-framework/api/pkg/operators/v2"
+	"github.com/pkg/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	patcher "github.com/openshift/windows-machine-config-operator/pkg/patch"
+)
+
+const (
+	upgradeableTrueReason   = "upgradeIsSafe"
+	upgradeableTrueMessage  = "The operator is safe for upgrade."
+	upgradeableFalseReason  = "upgradeIsNotSafe"
+	upgradeableFalseMessage = "The operator is currently processing sub-components."
+	// operatorConditionName is an environment variable set by OLM identifying the operator's OperatorCondition CR
+	OperatorConditionName = "OPERATOR_CONDITION_NAME"
+)
+
+// EnableOLMUpgrade is a wrapper that sets the Upgradeable condition to True to allow OLM to perform operator upgrades
+func EnableOLMUpgrade(c client.Client, watchNamespace string) error {
+	return set(c, watchNamespace, operators.Upgradeable, meta.ConditionTrue, upgradeableTrueReason,
+		upgradeableTrueMessage)
+}
+
+// DisableOLMUpgrade is a wrapper that sets the Upgradeable condition to False to prevent operator upgrades by OLM
+func DisableOLMUpgrade(c client.Client, watchNamespace string) error {
+	return set(c, watchNamespace, operators.Upgradeable, meta.ConditionFalse, upgradeableFalseReason,
+		upgradeableFalseMessage)
+}
+
+// set sets the given condition with the given values within the associated OperatorCondition.
+// No-op if operator is not managed by OLM
+func set(c client.Client, ocNamespace, conditionType string, status meta.ConditionStatus,
+	reason, message string) error {
+	// Check if operator is OLM-managed
+	if ocName, present := os.LookupEnv(OperatorConditionName); present {
```

Comment:
I take it that the env var not being present is not an error. Correct?

I also suggest checking for `!present` and returning nil. That way majority of the code will be less indented and easier to read.